nsxt/resource_nsxt_vpc.go

```diff
@@ -0,0 +1,626 @@
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"auto_snat": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "bool",
+							SdkFieldName: "AutoSnat",
+						},
+					},
+					"qos_config": {
+						Schema: schema.Schema{
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &metadata.ExtendedResource{
+								Schema: map[string]*metadata.ExtendedSchema{
+									"ingress_qos_profile_path": {
+										Schema: schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validatePolicyPath(),
+											Optional:     true,
+										},
+										Metadata: metadata.Metadata{
+											SchemaType:   "string",
+											SdkFieldName: "IngressQosProfilePath",
+										},
+									},
+									"egress_qos_profile_path": {
+										Schema: schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validatePolicyPath(),
+											Optional:     true,
+										},
+										Metadata: metadata.Metadata{
+											SchemaType:   "string",
+											SdkFieldName: "EgressQosProfilePath",
+										},
+									},
+								},
+							},
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "struct",
+							SdkFieldName: "QosConfig",
+							ReflectType:  reflect.TypeOf(model.GatewayQosProfileConfig{}),
+						},
+					},
+					"disable": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "bool",
+							SdkFieldName: "Disable",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "ServiceGateway",
+			ReflectType:  reflect.TypeOf(model.ServiceGateway{}),
+		},
+	},
+	"vpc_service_profile": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validatePolicyPath(),
+			Optional:     true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "VpcServiceProfile",
+		},
+	},
+	"load_balancer_vpc_endpoint": {
+		Schema: schema.Schema{
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"enabled": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "bool",
+							SdkFieldName: "Enabled",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "LoadBalancerVpcEndpoint",
+			ReflectType:  reflect.TypeOf(model.LoadBalancerVPCEndpoint{}),
+		},
+	},
+	"ip_address_type": {
```

Comment:
We could drop this entirely, it has only `IPV4` as valid value

Reply:
I think we can keep it here, when `ipv6` the enum list will grow automatically as result of resource evolution, and then no manual work will be required
however we might want to skip this in docs